### ISSUE TITLE
Repackage as an 'extension'? 

### ISSUE THREAD

**Issue by cmichelenstrofer** [SBuercklin/UnitfulChainRules.jl#14]
Could this be repackaged as an [extension](https://julialang.org/blog/2023/04/julia-1.9-highlights/#package_extensions), so that it is automatically added if both `Unitful` and `ChainRules` are used?

**Comment by SBuercklin** [comment]
Absolutely, in my opinion this is a perfect use case for package extensions. 

I don't have time to handle that right now, but I support migration into the core package and conditional loading from there. If you (or someone) ports it, let me know and I can put a note up on this repo. Might also be good to throw a deprecation warning if it's completely superfluous
